### ISSUE TITLE
No Proxy support for DynamoDB streams

### ISSUE THREAD

**Issue by ThinkBriK** [gravitational/teleport#2917]
DynamoDB streams do not honor HTTPS_PROXY env variable.
**What happened**:
Tried to set `HTTP_PROXY` and `HTTPS_PROXY` to use dynamoDB backend in a private environment, and I've got the error : 
```
ERRO [DYNAMODB]  "Poll streams returned with error: RequestError: send request failed\ncaused by: Post https://streams.dynamodb.eu-west-3.amazonaws.com/: dial tcp 52.46.68.12:443: connect: connection timed out." dynamo/shards.go:62
```

When using aws cli from the same VM : 
```
[~]$ export NO_PROXY=169.254.169.254
[~]$ export HTTP_PROXY=http://proxy.my.cloud:8080
[~]$ export HTTPS_PROXY=http://proxy.my.cloud:8080
[~]$ aws dynamodbstreams list-streams
{
    "Streams": [
        REDACTED
}

```

**What you expected to happen**:
Working HTTP Proxy support for DynamoDB streams

**How to reproduce it (as minimally and precisely as possible)**:
Add `HTTPS_PROXY` and `HTTP_PROXY` variable to the SystemD unit and `NO_PROXY=169.254.169.254` for metadata access in AWS.
Check logs

**Environment**:
- Teleport version (use `teleport version`): Teleport v4.0.4 git:v4.0.4-0-g1a2ed507 go1.12.1
- Tsh version (use `tsh version`): Teleport v4.0.4 git:v4.0.4-0-g1a2ed507 go1.12.1
- OS (e.g. from /etc/os-release): Centos 7

**Relevant Debug Logs If Applicable**

```
ERRO [DYNAMODB]  "Poll streams returned with error: RequestError: send request failed\ncaused by: Post https://streams.dynamodb.eu-west-3.amazonaws.com/: dial tcp 52.46.68.12:443: connect: connection timed out." dynamo/shards.go:62
```

**Comment by ThinkBriK** [comment]
Basically this issue prevent any private deployment with a proxified internet proxy : As DynamoDB Streams are not available via VPC endpoints and Teleport code doesn't support using a proxy, cluster members never reload their conf.

**Comment by zmb3** [comment]
Looks like this was fixed in #4271
